module/VuFind/src/VuFind/Search/Base/Explanation.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+/**
+ * Abstract explanation model.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Hebis Verbundzentrale 2023.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search_Base
+ * @author   Dennis Schrittenlocher <Dennis.Schrittenlocher@outlook.de>
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFind\Search\Base;
+
+use Laminas\Log\LoggerAwareInterface;
+use VuFind\Log\LoggerAwareTrait;
+use VuFindSearch\Service as SearchService;
+
+/**
+ * Abstract explanation model.
+ *
+ * This abstract class defines the methods for modeling an explanation in VuFind.
+ *
+ * @category VuFind
+ * @package  Search_Base
+ * @author   Dennis Schrittenlocher <Dennis.Schrittenlocher@outlook.de>
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+abstract class Explanation implements LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    /**
+     * Configuration
+     *
+     * @var \Laminas\Config\Config
+     */
+    protected $config;
+
+    /**
+     * Configuration file to read search settings from
+     *
+     * @var string
+     */
+    protected $searchIni = 'searches';
```

Comment:
I guess we could add Explanation.php in Search/Search2 that extends /Search/Solr/Explanations and overwrites $seachIni to "Search2". Or is there a better place for the config so it can be used for all explanations that use Solr?

Reply:
Enabled explain for search2.